include/nil/crypto3/math/polynomial/polynomial_dfs.hpp

```diff
@@ -761,6 +766,41 @@ namespace nil {
                 return os;
             }
 
+            template<typename FieldType>
+            static inline polynomial_dfs<typename FieldType::value_type> polynomial_sum(
+                    std::vector<math::polynomial_dfs<typename FieldType::value_type>> addends) {
+                using FieldValueType = typename FieldType::value_type;
+                std::size_t max_size = 0;
+                std::unordered_map<std::size_t, polynomial_dfs<FieldValueType>> size_to_part_sum;
+                for (auto& addend : addends) {
+                    max_size = std::max(max_size, addend.size());
+                    auto it = size_to_part_sum.find(addend.size());
+                    if (it == size_to_part_sum.end()) {
+                        size_to_part_sum[addend.size()] = std::move(addend);
+                    } else {
+                        it->second += addend;
+                        // Free the memory we are not going to use anymore.
+                        addend = math::polynomial_dfs<FieldValueType>();
+                    }
+                }
+
+                auto coef_result = polynomial<FieldValueType>(max_size, FieldValueType::zero());
+                for (auto& [_, partial_sum] : size_to_part_sum) {
+                    FieldValueType omega = unity_root<FieldType>(partial_sum.size());
```

Comment:
I did not understand these loop. You just needed to write something like coef_result += polynomial<FieldValueType>(partial_sum.coefficients());
partial_sum = math::polynomial_dfs<FieldValueType>();